test/TitleEscrowCloner.js

```diff
@@ -0,0 +1,113 @@
+    expect(escrow1Beneficiary).to.be.equal(beneficiary1.address);
+    expect(escrow1Holder).to.be.equal(beneficiary1.address);
+    expect(escrow1TokenRegistry).to.be.equal(ERC721Address);
+
+    const escrow2Receipt = await (
+      await cloner.deployNewTitleEscrow(ERC721Address, beneficiary2.address, beneficiary2.address)
+    ).wait();
+    console.log(escrow2Receipt.events[1].args.escrowAddress);
+    const escrow2Instance = await TitleEscrowCloneableFactory.attach(escrow2Receipt.events[1].args.escrowAddress);
+    const escrow2Beneficiary = await escrow2Instance.beneficiary();
+    const escrow2Holder = await escrow2Instance.holder();
+    const escrow2TokenRegistry = await escrow2Instance.tokenRegistry();
+    expect(escrow2Beneficiary).to.be.equal(beneficiary2.address);
+    expect(escrow2Holder).to.be.equal(beneficiary2.address);
+    expect(escrow2TokenRegistry).to.be.equal(ERC721Address);
+  });
+
+  it("should allow instances of TitleEscrow to have the function transferToNewEscrow", async () => {
+    const { events } = await (
+      await ERC721Instance.deployNewTitleEscrow(ERC721Address, beneficiary1.address, beneficiary1.address)
+    ).wait();
+    const { escrowAddress } = events[1].args;
+
+    await ERC721Instance["safeMint(address,uint256)"](escrowAddress, SAMPLE_TOKEN_ID);
+
+    const escrowInstance = await TitleEscrowCloneableFactory.attach(escrowAddress);
+    const receipt = await (
+      await escrowInstance.connect(beneficiary1).transferToNewEscrow(beneficiary2.address, holder2.address)
+    ).wait();
+    const titleCededLog = receipt.events.find((log) => log.event === "TitleCeded");
+    const newAddress = titleCededLog.args._to; // eslint-disable-line no-underscore-dangle
+    const ownerOnRegistry = await ERC721Instance.ownerOf(SAMPLE_TOKEN_ID);
+
+    expect(escrowAddress).not.to.be.equal(newAddress);
+    expect(newAddress).to.be.equal(ownerOnRegistry);
+
+    const newEscrowInstance = await TitleEscrowCloneableFactory.attach(ownerOnRegistry);
+    const escrowBeneficiary = await newEscrowInstance.beneficiary();
+    const escrowHolder = await newEscrowInstance.holder();
+    const escrowTokenRegistry = await newEscrowInstance.tokenRegistry();
+    expect(escrowBeneficiary).to.be.equal(beneficiary2.address);
+    expect(escrowHolder).to.be.equal(holder2.address);
+    expect(escrowTokenRegistry).to.be.equal(ERC721Address);
+  });
+
+  // TODO: test that implementation's onERC721Received cannot be called
+  it("should not be able to call onERC721Received on the implementation", async () => {
+    const implAddr = await cloner.titleEscrowImplementation();
+    const implInstance = await TitleEscrowCloneableFactory.attach(implAddr);
+
+    const onReceived = implInstance.onERC721Received(implAddr, implAddr, SAMPLE_TOKEN_ID, "0x");
+
+    await expect(onReceived).to.be.revertedWith("Only tokens from predefined token registry can be accepted");
+  });
```

Comment:
i don't recall what i was thinking when i wrote that... it does seem a bit weird when you put it this way
(did i even write this? 🤔)

i think the idea might be to check that the implementation should not keep any state that might affect proxies, but i dont really see how that could happen with or without this test

Reply:
I'm thinking it probably won't affect the proxies even if the states in the implementation got messed around with because the proxies are calling via `delegateCall`.🤔